js/main.js

```diff
@@ -0,0 +1,119 @@
+function Finder(status) {
+  this.status = 'home-page';
+}
+
+var holiday = new Finder();
+
+var homePage = document.querySelector('.home-page');
+var searchPage = document.querySelector('.search-page');
+var resultPage = document.querySelector('.result-page');
+
+var findButton = document.querySelector('.home-search');
+var submit = document.querySelector('.submit');
+var dataOutline = document.querySelector('.data-outline');
+var back = document.querySelector('.back');
+
+var searchBox = document.querySelector('#search-box');
+
+var backHome = ['search-page'];
+
+var main = document.querySelector('main');
+
+Finder.prototype.changePage = function (event) {
+  if (event.target === back && holiday.status === 'home-page') {
+    return;
+  }
+
+  if (event.target === searchBox) {
+    return;
+  }
+
+  homePage.className = 'home-page hidden';
+  searchPage.className = 'search-page hidden';
+  resultPage.className = 'result-page hidden';
+  if (event.target === findButton) {
+    searchPage.className = 'search-page';
+    holiday.status = 'search-page';
+  } else if (event.target === submit) {
+    dataOutline.innerHTML = '';
+    holiday.renderResults();
+    resultPage.className = 'result-page';
+    holiday.status = 'result-page';
+  } else if (event.target === back) {
+    for (var i = 0; backHome.length > i; i++) {
+      if (backHome[i] === holiday.status) {
+        homePage.className = 'home-page';
+        holiday.status = 'home-page';
+      }
+    }
+    if (holiday.status === 'result-page') {
+      searchPage.className = 'search-page';
+      holiday.status = 'search-page';
+    }
+  }
+};
+
+Finder.prototype.renderResults = function () {
+  var xhr = new XMLHttpRequest();
+  xhr.open('GET', 'https://holidayapi.com/v1/holidays?pretty&key=c1e47cbf-5b67-44b0-8359-4fe2afc3ae3a&country=US&year=2020');
+  xhr.responseType = 'json';
+  xhr.addEventListener('load', function () {
+    var i = 0;
+    var addy = [];
+    while (xhr.response.holidays.length > i) {
+      if (searchBox.value === xhr.response.holidays[i].date) {
+        addy.push(i);
+        i++;
+      } else {
+        i++;
+      }
+    }
```

Comment:
A `for` loop would be cleaner than a `while` loop, since you already have all the pieces.